numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -2541,6 +2542,127 @@ def add_newdoc(place, name, doc):
 
     """)
 
+add_newdoc('numpy.core.umath', 'matmul',
+    """
+    Matrix product of two arrays.
+
+    Parameters
+    ----------
+    x1, x2 : array_like
+        Input arrays, scalars not allowed.
+    out : ndarray, optional
+        Output argument. This must have the exact kind that would be returned
+        if it was not used. In particular, it must be C-contiguous and have
+        both the right type and the right dtype. This is a performance feature.
```

Comment:
Wait, why is `matmul` special here? All the linalg gufuncs handle this just fine.

Reply:
```
>>> np.linalg.svd(np.arange(12).reshape(3, 4), out=<anything>)
TypeError: svd() got an unexpected keyword argument 'out'
```

Reply:
`np.linalg.svd` isn't the gufunc - the gufunc is hidden internally.

Reply:
I copied that text from the 1.15.4, pre-ufunc version. There should be no reason for this ufunc to be different from any other ufunc now.

Reply:
adopting text from other ufuncs